app/scripts/views/NodeViews/Base.js

```diff
@@ -1,5 +1,61 @@
 define(['backbone', 'jqueryuidraggable', 'bootstrap', 'Hammer'], function(Backbone, jqueryuidraggable, bootstrap, Hammer) {
 
+  var itemsToShow = 10, minItemsNumber = 1, maxItemsNumber = 1;
+  function changeItemsNumber (e) {
+      itemsToShow = parseInt(this.$el.find('input.shown-items').val());
+      this.$el.find('li.array-item').filter(function () {
+          return  parseInt($(this).attr("data-index")) >= itemsToShow;
+      }).addClass('hidden');
+
+      this.$el.find('li.array-item').filter(function () {
+          return  parseInt($(this).attr("data-index")) < itemsToShow;
+      }).removeClass('hidden');
+
+      var dotsItem = this.$el.find('li.dots-item');
+      if (this.model.get('arrayItems').length > itemsToShow) {
+          dotsItem.removeClass('hidden');
+      }
+      else {
+          dotsItem.addClass('hidden');
+      }
+
+      this.$el.find('span.shown-items').html('of ' + maxItemsNumber +
+          (itemsToShow > 1 ? ' are' : ' is') + ' shown');
+
+      e && e.stopPropagation();
+  }
+
+  function keyDown (e) {
+      console.log(e.keyCode);
+      var zeroCode = 48, nineCode = 57,
+          rightSidedZeroCode = 96, rightSidedNineCode = 105;
+      // backspace, shift, ctrl, 4 arrows, delete
+      var allowedKeyCodes = [8, 16, 17, 37, 38, 39, 40, 46];
```

Comment:
allow this buttons as well to give a user a possibility of moving cursor, selecting and deleting typed number